### DISCUSSION TITLE
Background task blocks response from being sent

### DISCUSSION THREAD

**Discussion by sgondala** [fastapi/fastapi#11675] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [x] I commit to help with one of those options 👆

### Example Code

```python
async def sleep_10_fn():
    time.sleep(10)
    print("Done waiting")
    return


@app.post("/long_waiting_task")
async def long_waiting_task(background_tasks: BackgroundTasks):
    background_tasks.add_task(sleep_10_fn)

    return PlainTextResponse(
        content="OK",
        status_code=status.HTTP_200_OK,
    )
```

### Description

In this code, I don't hear response back until background task finishes. I understand that background task is blocking the whole thread so nothing else can execute, but shouldn't background task start "after" the response is sent? 

Why is background task being blocking causing problems with sending response?

### Operating System

macOS

### FastAPI Version

0.110.0

### Pydantic Version

2.7.1

### Python Version

3.10.13

**Comment by JavierSanchezCastro** [comment]
You use a block function (time.sleep) inside an async function.
If you change to asyncio.sleep, is gonna work as expected

```python3
from fastapi import FastAPI, BackgroundTasks, status
from fastapi.responses import PlainTextResponse
import time
import asyncio

app = FastAPI()

async def sleep_10_fn():
    await asyncio.sleep(10)
    print("Done waiting")
    return


@app.post("/long_waiting_task")
async def long_waiting_task(background_tasks: BackgroundTasks):
    background_tasks.add_task(sleep_10_fn)

    return PlainTextResponse(
        content="OK",
        status_code=status.HTTP_200_OK,
    )
```

**Reply by sgondala** [reply]
I used a blocking function to illustrate that the function takes time. 

My question was more around - if background task executes after response is sent, why is response being blocked by background task execution?

**Reply by Kludex** [reply, marked as answer]
Ah, it's because you are sending a Response object. If you send a non-response object in the return, should work.
